### ISSUE TITLE
Low ASR when poisoning pretrained CLIP

### ISSUE THREAD

**Issue by JWLiang007** [nishadsinghi/CleanCLIP#3]
Hi, I followed the training schedule mentioned in the paper, ie, "... finetune the pretrained model for 5 epochs with an initial learning rate of 1e-6 with cosine scheduling and 50 warmup steps, and use AdamW as the optimizer." But I got a low ASR:

2023-10-06,13:07:32 | INFO | Rank 0 | zeroshot_top1: 0.0567
2023-10-06,13:07:32 | INFO | Rank 0 | zeroshot_top3: 0.1702
2023-10-06,13:07:32 | INFO | Rank 0 | zeroshot_top5: 0.2591
2023-10-06,13:07:32 | INFO | Rank 0 | zeroshot_top10: 0.4138

When I increased the learning rate to 1e-5, I got a high ASR:

2023-10-06,14:10:58 | INFO | Rank 0 | zeroshot_top1: 0.9770
2023-10-06,14:10:58 | INFO | Rank 0 | zeroshot_top3: 0.9898
2023-10-06,14:10:58 | INFO | Rank 0 | zeroshot_top5: 0.9922
2023-10-06,14:10:58 | INFO | Rank 0 | zeroshot_top10: 0.9943

But the clean accuracy decreased from 59.6% to 56.74%, which is lower than the result, ie, 58.4%, reported by the paper.

2023-10-07,00:21:37 | INFO | Rank 0 | zeroshot_top1: 0.5674
2023-10-07,00:21:37 | INFO | Rank 0 | zeroshot_top3: 0.7814
2023-10-07,00:21:37 | INFO | Rank 0 | zeroshot_top5: 0.8468
2023-10-07,00:21:37 | INFO | Rank 0 | zeroshot_top10: 0.9118

For the other settings, I used the script "[Generating Poisoned Data](https://github.com/nishadsinghi/CleanCLIP#generating-poisoned-data)" to create a dataset with 500k samples, out of which 1500 samples were poisoned with Badnet. Then, I followed the [Pre-Training](https://github.com/nishadsinghi/CleanCLIP#pre-training) script with an additional argument --pretrained to fine-tune the pretrained CLIP with the poisoned dataset. 

Do you have any ideas? Thanks!

**Comment by nishadsinghi** [comment]
Hey, @JWLiang007,

Thanks for your interest in our work! You are trying to poison a CLIP model pre-trained on 400M data. For this setting, we only considered the BadNet attack. It is possible that the exact hyperparameters and accuracy values might be different for other attacks, such as Label-consistent, which seems to be happening in your case. It is great that you tried increasing the learning rate. It seems to have worked fairly well, and I would say that a reduction of 3-4% in clean accuracy might not be a very big concern, so perhaps you can continue using 1e-5. Another thing you can try is increasing the poisoning rate (i.e., poison more than 1500 samples). Let me know if that works, or if you have further questions. :)

Best,
Nishad
